q2_vsearch/_stats.py

```diff
@@ -0,0 +1,131 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import pkg_resources
+import subprocess
+import pandas as pd
+from multiprocessing import Pool, cpu_count
+
+from q2_types.per_sample_sequences import (
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt)
+import q2templates
+
+TEMPLATES = pkg_resources.resource_filename('q2_vsearch', 'assets')
+
+
+def _get_stats(cmds) -> None:
+    cmd, cmd2 = cmds
+    process1 = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+    process2 = subprocess.Popen(cmd2, stdin=process1.stdout)
+
+    while True:
+        process1.poll()
+        process2.poll()
+        if process1.returncode is not None and process2.returncode is not None:
+            process1.communicate(timeout=15)
+            break
+
+
+def _build_cmds(output_dir: str, filelist, direction='forward'):
+    datafiles = {direction: {}}
+    cmd = ['zcat'] + filelist
```

Comment:
I am wary of relying on a system tool like this. Is it (or an alternative) available on conda? If so, please update and specify the dependency in the build recipe. If it is not, please refactor to use python stdlib. Also, I am a bit confused about _why_ this is here --- it looks like the `zcat` results are duplicated twice, but it doesn't look like the results are used anywhere - can you please comment on that?

Reply:
Hi thermokarst,
You are right about using `zcat` without being sure if it is installed.

zcat is used to get all reads over all samples of one direction as a pipe (line 25) which is used by the vsearch-command as input from stdin `-` (lines 26, 42). vsearch can handle only one gz-file ;(

Nonetheless, I refactored the code to use a python solution and also made it possible to reduce I/O when using more cores.